elisp/guix-devel.el

```diff
@@ -338,6 +338,46 @@ Each rule should have a form (SYMBOL VALUE).  See `put' for details."
   (add-before 'guix-devel-indent-modify-phases-keyword-2))
 
 
+;;; String inside sexp
+
+(require 'edit-indirect nil t)
+
+(defconst guix-devel-code-block-syntax
+  '("description" "synopsis"))
+
+(defun guix-devel-code-block-position ()
+  "Return begining and end of a string inside
+`guix-devel-code-block-syntax' sexp."
+  (save-excursion
+    (re-search-backward (rx-to-string
+                         `(and (syntax open-parenthesis)
+                               (or ,@guix-devel-code-block-syntax)
+                               not-wordchar)
+                         t))
```

Comment:
If there is no synopsis/description before the current position, this
search fails, so I think it would be good to make a message about it.
Something like:

```elisp
(or (re-search-backward guix-devel-code-block-regexp nil t)
      (user-error "The point should be inside 'description' or 'synopsis'"))
```

Reply:
`(re-search-backward guix-devel-code-block-regexp nil t)` will be almost always true.

```emacs-lisp
(define-public dog
  (package
    (name "")
    (version "")
    (source origin...)
    (build-system )
    (home-page "")
    (synopsis "")
    (description "Will insert text here.")
    (license )))

(define-public cat
  (package
    (name "")
    (version "")
    (source origin...)
    (build-system ) ; (point) is here.  Let's hit <C-'> and jump to dog's `description`
    (home-page "")
    (synopsis "")
    (description "")
    (license )))
```

Reply:
Yeah, I know, but not always: it will fail if the point is before synopsis/description of the first package of a file, so I think it is better to handle this error anyway

Reply:
As a workaround I used `narrow-to-defun` which works for all cases.